File: Algorithm/python/BAEKJOON/CLASS 1/10250.py
# TRY 2
T = int(input())
for t in range(T):
    H, W, N = map(int, input().split(" "))

    room_list = []

    for w in range(1, W + 1):
        for h in range(1, H + 1):
            room_list += [h * 100 + w]
    answer = room_list[N - 1]
    print(answer)

# # TRY 1
# T = int(input())
# for t in range(T):
#     H, W, N = map(int, input().split(" "))
#
#    room_list = []
#    for w in range(1, W + 1):
#        for h in range(1, H + 1):
#            room_list += [str(h) + str(w).zfill(len(str(W)))]
#
#    answer = int(room_list[N-1])
#    print(answer)
# # 다음과 같이 진행하면 W의 길이가 1인 경우 11과 같이 출력

File: Algorithm/javascript/BAEKJOON/CLASS 1/10250.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", (line) => {
  if (!line) {
    rl.close();
  } else {
    input.push(line);
  }
}).on("close", () => {
  solution(input);
  process.exit();
});

function solution(input) {
  let T = input[0];
  for (let i = 1; i <= T; i++) {
    let H = input[i].split(" ")[0];
    let W = input[i].split(" ")[1];
    let N = input[i].split(" ")[2];

    roomList = [];
    for (let w = 1; w <= W; w++) {
      for (let h = 1; h <= H; h++) {
        answer = h * 100 + w;
        roomList.push(answer);
      }
    }
    console.log(roomList[N - 1]);
  }
}
